src/vellum/workflows/triggers/base.py

```diff
@@ -0,0 +1,207 @@
+                elif isinstance(item, Graph):
+                    # Trigger >> {Graph1, Graph2}
+                    graph_items.append(item)
+                    for entrypoint in item.entrypoints:
+                        trigger_edges.append(TriggerEdge(self, entrypoint))
+                else:
+                    raise TypeError(
+                        f"Cannot connect trigger to {type(item).__name__}. " f"Expected BaseNode or Graph in set."
+                    )
+
+            result_graph = Graph.from_trigger_edges(trigger_edges)
+
+            for graph_item in graph_items:
+                result_graph._extend_edges(graph_item.edges)
+                result_graph._terminals.update(graph_item._terminals)
+                for existing_trigger_edge in graph_item._trigger_edges:
+                    if existing_trigger_edge not in result_graph._trigger_edges:
+                        result_graph._trigger_edges.append(existing_trigger_edge)
+
+            return result_graph
+
+        elif isinstance(other, Graph):
+            # Trigger >> Graph
+            edges = [TriggerEdge(self, entrypoint) for entrypoint in other.entrypoints]
+            result_graph = Graph.from_trigger_edges(edges)
+            # Also include the edges from the original graph
+            result_graph._extend_edges(other.edges)
+            result_graph._terminals = other._terminals
+            return result_graph
+
+        elif isinstance(other, type) and issubclass(other, BaseNodeClass):
+            # Trigger >> Node
+            edge = TriggerEdge(self, other)
+            return Graph.from_trigger_edge(edge)
+
+        else:
+            raise TypeError(
+                f"Cannot connect trigger to {type(other).__name__}. " f"Expected BaseNode, Graph, or set of BaseNodes."
+            )
+
+    def __rrshift__(self, other: Any) -> "Graph":
+        """
+        Prevent Node >> Trigger syntax.
+
+        Triggers must always be at the start of a graph path, never as targets.
+        This method raises a TypeError to catch invalid usage.
+
+        Raises:
+            TypeError: Always, as this operation is not allowed
+        """
+        raise TypeError(
+            f"Cannot create edge targeting trigger {self.__class__.__name__}. "
+            f"Triggers must be at the start of a graph path, not as targets. "
+            f"Did you mean: {self.__class__.__name__}() >> {other.__name__ if hasattr(other, '__name__') else other}?"
+        )
+
+    def __hash__(self) -> int:
+        """Hash based on trigger class for set/dict operations."""
+        return hash(self.__class__)
+
+    def __eq__(self, other: object) -> bool:
+        """Triggers are equal if they're the same class."""
+        if not isinstance(other, BaseTrigger):
+            return False
+        return self.__class__ == other.__class__
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}()"
+
+    def get_trigger_inputs(self) -> Dict[str, Any]:
```

Comment:
[nit] would be nice to say `get_trigger_attributes`, would be nice to keep consistent

Reply:
oh actually, [blocking] we don't want any top level attributes here in the base as it could conflict with the trigger's own attributes. What would we need this for? Downstream nodes will use Trigger data by doing things like:
```
class MyPrompt(BaseNode):
     blocks = [SlackTrigger.message]
```